pkg/asset/installconfig/gcp/validation.go

```diff
@@ -133,3 +134,35 @@ func validateMachineNetworksContainIP(fldPath *field.Path, networks []types.Mach
 	}
 	return field.ErrorList{field.Invalid(fldPath, subnetName, fmt.Sprintf("subnet CIDR range start %s is outside of the specified machine networks", ip))}
 }
+
+//ValidateEnabledServices gets all the enabled services for a project and validate if any of the required services are not enabled.
+func ValidateEnabledServices(client API, ctx context.Context, project string) error {
+	services := sets.NewString("compute.googleapis.com",
+		"cloudapis.googleapis.com",
+		"cloudresourcemanager.googleapis.com",
+		"dns.googleapis.com",
+		"iam.googleapis.com",
+		"iamcredentials.googleapis.com",
+		"servicemanagement.googleapis.com",
+		"serviceusage.googleapis.com",
+		"storage-api.googleapis.com",
+		"storage-component.googleapis.com")
+	//the Api require the type of parent for which the services are fetched. Here the parent is projects/
+	parent := "projects/"
+	projectServices, err := client.GetEnabledServices(ctx, parent+project)
```

Comment:
GetEnabledServices should handle the project_id to `projects/<project id>`, I don't think it's good to move that responsibility to the caller.